### PR TITLE
refactor(dlint): replace `markdown` with `pulldown-cmark`

### PR DESCRIPTION
**Pull request by magurotuna**
This PR replaces `markdown` with `pulldown-cmark`. Both are used as markdown parser for printing markdown in pretty format to the terminal. Both work fine for this purpose, but `pulldown-cmark` is preferable because it already exists in the dependency tree of Deno.

To properly handle nested decorations like "bar" in "**foo ~~bar~~ baz**", I had to work with ANSI color codes directly, rather than using an external crate.

Ref #765, https://github.com/denoland/deno/pull/11244

### Demo

no-this-alias

image

no-deprecated-deno-api

image

### CODE REVIEW

**Comment by bartlomieju** [comment]
Thanks @magurotuna, this looks good to me, the colors for highlighting look really good. @dsherret do you have any remarks?

File: examples/dlint/color.rs [hunk 1]
```diff
@@ -1,195 +1,318 @@
...
+use if_chain::if_chain;
+use pulldown_cmark::{Options, Parser, Tag};
 use std::convert::TryFrom;
 use swc_ecmascript::parser::token::{Token, Word};
 
-pub trait Colorize {
-  fn colorize(self) -> String;
+pub fn colorize_markdown(input: &str) -> String {
+  let mut options = Options::empty();
+  options.insert(Options::ENABLE_STRIKETHROUGH);
```
Comment (dsherret) [inline examples/dlint/color.rs:10]: Maybe we don't need more options for dlint, but what I did in dprint was:

```rust
options.insert(Options::ENABLE_TABLES);
options.insert(Options::ENABLE_FOOTNOTES);
options.insert(Options::ENABLE_STRIKETHROUGH);
options.insert(Options::ENABLE_TASKLISTS);
```

**Review by dsherret — APPROVED** [review]
LGTM too

File: examples/dlint/color.rs (hunk 1 above)
Comment (magurotuna) [reply examples/dlint/color.rs:10]: thanks for the info. We'll enable other options once it's needed :)

**Pull request merged**